opentasks-provider/src/main/java/org/dmfs/provider/tasks/model/adapters/DateTimeIterableFieldAdapter.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.provider.tasks.model.adapters;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+
+import org.dmfs.iterables.EmptyIterable;
+import org.dmfs.iterables.Split;
+import org.dmfs.iterables.decorators.DelegatingIterable;
+import org.dmfs.jems.iterable.decorators.Mapped;
+import org.dmfs.jems.iterable.generators.ConstantGenerator;
+import org.dmfs.jems.single.elementary.Reduced;
+import org.dmfs.provider.tasks.utils.iterable.Alternating;
+import org.dmfs.rfc5545.DateTime;
+
+import java.util.TimeZone;
+
+
+/**
+ * Knows how to load and store {@link Iterable}s of {@link DateTime} values from a {@link Cursor} or {@link ContentValues}.
+ *
+ * @param <EntityType>
+ *         The type of the entity the field belongs to.
+ *
+ * @author Marten Gajda
+ */
+public final class DateTimeIterableFieldAdapter<EntityType> extends SimpleFieldAdapter<Iterable<DateTime>, EntityType>
+{
+    private final String mDateTimeListFieldName;
+    private final String mTimeZoneFieldName;
+
+
+    /**
+     * Constructor for a new {@link DateTimeIterableFieldAdapter}.
+     *
+     * @param datetimeListFieldName
+     *         The name of the field that holds the {@link DateTime} list.
+     * @param timezoneFieldName
+     *         The name of the field that holds the time zone name.
+     */
+    public DateTimeIterableFieldAdapter(String datetimeListFieldName, String timezoneFieldName)
```

Comment:
Could we start using `@NonNull` and `@Nullable` everywhere? The project depends heavily currently on values being `null`, so it could make it clearer. IDE can also show warnings. Also if we happen to migrate to Kotlin, those annotations are very helpful for the interoperatilbity.

In this class specifically, one field is effectively nullable, the other is not, so it would help understand the null checks below.

Reply:
Yeah, ideally we get rid of all the `null` stuff. This class in particular could be split into two, one which reads and writes the timestamps and a decorator which handles the time zones.

Nevertheless, using the annotations is still a good idea.